### DISCUSSION TITLE
ERROR [vpn] kernel does not support Wireguard

### DISCUSSION THREAD

**Discussion by asbor** [qdm12/gluetun#1963] [Q&A] [answered]
Hi, im not exactly sure how this happened, but i had this up and running for several days, and now out of the blue i am not able to establish a connection again.  and im getting the message 
- **ERROR [vpn] kernel does not support Wireguard**

To clarify, im running the Gluetun VPN client on fresh installation of Unraid 6.12.4
      
Application Type | Docker
-- | --
Categories | Security, Network:VPN
Added | Jul 23, 2021
Repository | qmcgaw/gluetun
DockerHub Stars: | 201
Last Update: | Nov 8, 2023

anyone have any ideas on how to solve this ? I couldn't find any information related to this anywhere.

**Comment by asbor** [comment, marked as answer]
Somehow the issue solved itself. It looks like the issue might have been caused by rebuilding the dependent applications while they were running, instead of turning them off. So the solution was actually to make a small change in the vpn, while everything was running, this reverted the configurations back. and since then, for every configuration which is done, turn everything off prior to the configuration, and it appears to work as expected.

**Comment by argonan0** [comment]
I've seen this many times and it is a misleading issue since typically the Unraid kernel does support Wireguard and it is functional.

Still, it would be nice if either a better message was shown or it could self-heal more elegantly.

**Comment by mattdale77** [comment]
I just encountered this with a ubuntu host. I use kernelspace wireguard. My solution was to comment this out to revert to userspace and rebuild the container and then remove the comment, reverting to kernelspace and it has come up healthy. It's possible that other arbitrary configuration changes would work but this is what worked for me
